Compile fails on ppc64el due to `constexpr long double`

### Operating System

Debian sid, ppc64el

### OpenRCT2 build

OpenRCT2, v0.4.0

### Describe the issue

After updating the Debian packages for OpenRCT2 to v0.4.0, the build for ppc64el failed due to what is apparently a weird gcc bug on that architecture related to how a `long double` is represented:

```
In file included from /<<PKGBUILDDIR>>/src/openrct2/ride/ShopItem.h:12,
                 from /<<PKGBUILDDIR>>/src/openrct2/ride/ShopItem.cpp:10:
/<<PKGBUILDDIR>>/src/openrct2/ride/ShopItem.cpp:29:41:   in ‘constexpr’ expansion of ‘operator""_GBP(2.99999999999999999999999999999998e-1l)’
/<<PKGBUILDDIR>>/src/openrct2/ride/../common.h:98:18: error: ‘(2.99999999999999999999999999999998e-1l * 1.0e+1l)’ is not a constant expression
   98 |     return money * 10;
      |            ~~~~~~^~~~
```

There are some other reports of similar issues online:
https://github.com/mamedev/mame/issues/3157
https://bugzilla.mozilla.org/show_bug.cgi?id=1416170
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=26374

I'm not sure what the correct fix for this would be; maybe removing the `constexpr` from `constexpr money32 operator"" _GBP(long double money) noexcept` as defined in src/openrct2/common.h? Or if a normal `double` would be sufficient, just use that?

There are also several variations on a warning about a narrowing conversion:

```
/<<PKGBUILDDIR>>/src/openrct2/ride/ShopItem.cpp:29:41: error: narrowing conversion of ‘operator""_GBP(2.99999999999999999999999999999998e-1l)’ from ‘money32’ {aka ‘int’} to ‘money16’ {aka ‘short int’} [-Werror=narrowing]
   29 |     /* ShopItem::Balloon */          {  0.30_GBP, 1.40_GBP,   1.40_GBP,  1.40_GBP,   0.90_GBP,      SPR_SHOP_ITEM_BALLOON,             { STR_SHOP_ITEM_PRICE_LABEL_BALLOON,                STR_SHOP_ITEM_SINGULAR_BALLOON,             STR_SHOP_ITEM_PLURAL_BALLOON,               STR_SHOP_ITEM_INDEFINITE_BALLOON,               STR_SHOP_ITEM_DISPLAY_BALLOON             }, SHOP_ITEM_FLAG_IS_SOUVENIR,                           Litter::Type::Rubbish,          0,                ShopItem::None,             PeepThoughtType::BalloonMuch,           PeepThoughtType::Balloon            },
      |                                         ^~~~~~~~
```

Previously, OpenRCT2 0.3.5.1 compiled just fine; full build log is at https://buildd.debian.org/status/fetch.php?pkg=openrct2&arch=ppc64el&ver=0.4.0%2Bds-1&stamp=1650977105&raw=0

### Area(s) with issue?

Building the game

### Steps to reproduce

1. Compile the latest release on a ppc64el box.

### Attachments

_No response_

Compilation is still broken.  Now there only appears to be one error left:

```
In file included from /<<PKGBUILDDIR>>/src/openrct2/ride/ShopItem.h:12,
                 from /<<PKGBUILDDIR>>/src/openrct2/ride/ShopItem.cpp:10:
/<<PKGBUILDDIR>>/src/openrct2/ride/ShopItem.cpp:29:41:   in ‘constexpr’ expansion of ‘operator""_GBP(2.99999999999999999999999999999998e-1l)’
/<<PKGBUILDDIR>>/src/openrct2/ride/../common.h:92:18: error: ‘(2.99999999999999999999999999999998e-1l * 1.0e+1l)’ is not a constant expression
   92 |     return money * 10;
      |            ~~~~~~^~~~
make[3]: *** [CMakeFiles/libopenrct2.dir/build.make:4265: CMakeFiles/libopenrct2.dir/src/openrct2/ride/ShopItem.cpp.o] Error 1
```
(https://launchpadlibrarian.net/654385319/buildlog_ubuntu-jammy-ppc64el.openrct2_0.4.3-develop-0~4200-202303032113-g463234e59~ubuntu22.04.1_BUILDING.txt.gz)

Hopefully this is a simple fix.

--- a/src/openrct2/common.h
+++ b/src/openrct2/common.h
@@ -87,9 +87,17 @@ using money64 = fixed64_1dp;
 #define FIXED_1DP(whole, fraction) FIXED_XDP(1, whole, fraction)
 #define FIXED_2DP(whole, fraction) FIXED_XDP(10, whole, fraction)
 
+// For a user defined floating point literal, the parameter type must be a
+// `long double` which is problematic on ppc64el, as the architecture uses a
+// pair of `doubles` to represent that type. This cannot be converted to a
+// `constexpr`. As a workaround, statically cast the `long double` down to a
+// `double`. All of the uses of _GBP constants fit just fine, and if anyone
+// really tries to use a gigantic constant that can't fit in a double, they are
+// probably going to be breaking other things anyways.
+// For more details, see https://gcc.gnu.org/bugzilla/show_bug.cgi?id=26374
 constexpr money64 operator"" _GBP(long double money) noexcept
 {
-    return money * 10;
+    return static_cast<double>(money) * 10;
 }
 
 constexpr money64 ToMoney64FromGBP(int32_t money) noexcept

